lighthouse-core/report/templates/report-template.html

```diff
@@ -46,6 +46,13 @@
             <a href="#" data-action="save-json">Save as JSON...</a>
           </ul>
         </span>
+        <span class="rerun-section">
+          <button class="report-body__icon rerun-button js-rerun-button"></button>
+          <div class="rerun-popup js-rerun-popup" status="idle">
+            <p class="rerun-message">Rerunning Lighthouse</p>
+            <a class="rerun-report-link js-rerun-report-link" href="/?id=rerun">New Report</a>
```

Comment:
Let's just go with the button rotating while its running. After its finished, load the new report in the same page. Also add a title to the button.

Optionally, gray out the background of the report while its running.